### PR TITLE
Initial Version with Data Preprocessing

### PR DESCRIPTION
**Pull request by tkilias**
Derived from LargeDataTransformer.py in https://github.com/exasol/data-science-examples/pull/10/

### CODE REVIEW

File: tests/integration_tests/test_table_preprocessor.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    CREATE SCHEMA IF NOT EXISTS "TARGET_SCHEMA"
+    """)
+    c.execute("""
+    CREATE OR REPLACE TABLE "SOURCE_SCHEMA"."SOURCE_TABLE" (
+        "CATEGORY" VARCHAR(1000),
+        "NUMERICAL" FLOAT
+    )
+    """)
+    c.execute("""INSERT INTO "SOURCE_SCHEMA"."SOURCE_TABLE" VALUES ('A',1),('B',2);""")
+    source_schema = "SOURCE_SCHEMA"
```
Comment (AnastasiiaSergienko) [inline tests/integration_tests/test_table_preprocessor.py:25]: Maybe we can also use these variables in queries above

**Review by AnastasiiaSergienko — COMMENTED** [review]
Reviewed with a few questions

**Comment by tkilias** [comment]
Changed branch new PR is #5 most of the concerns of the review were applied

File: exasol_data_science_utils_python/preprocessing/column_preprocessor.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    def _get_table_alias(self, target_schema: str, source_schema: str, source_table: str, source_column: str,
+                         prefix: str):
+        alias = f'"{target_schema}_{source_schema}_{source_table}_{source_column}_{prefix}"'
+        return alias
+
+    def _get_target_table_name(self, target_schema: str, source_schema: str, source_table: str, source_column: str,
+                               prefix: str):
+        return f'"{target_schema}"."{source_schema}_{source_table}_{source_column}_{prefix}"'
+
+    def _get_table_qualified(self, source_schema: str, source_table: str):
```
[the remark this answers is not in the document]
Comment (tkilias) [reply exasol_data_science_utils_python/preprocessing/column_preprocessor.py]: Solved by using classes for Schema, Table and Columns

File: exasol_data_science_utils_python/preprocessing/column_preprocessor.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        pass
+
+    @abstractmethod
+    def create_from_clause_part(self, source_schema: str, source_table: str, source_column: str,
+                                input_schema: str, input_table: str,
+                                target_schema: str) -> List[str]:
+        pass
+
+    @abstractmethod
+    def create_select_clause_part(self, source_schema: str, source_table: str, source_column: str,
```
[the remark this answers is not in the document]
Comment (tkilias) [reply exasol_data_science_utils_python/preprocessing/column_preprocessor.py]: Solved by using classes for Schema, Table and Columns

File: exasol_data_science_utils_python/preprocessing/normalization/min_max_scaler.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import textwrap
+from typing import List
+
+from exasol_data_science_utils_python.preprocessing.column_preprocessor import ColumnPreprocessor
+
+MIN_MAX_SCALAR_PARAMETER_TABLE_PREFIX = "MIN_MAX_SCALAR_PARAMETERS"
```
[the remark this answers is not in the document]
Comment (tkilias) [reply exasol_data_science_utils_python/preprocessing/normalization/min_max_scaler.py:9]: A column processor can potentially have multiple additional tables with specific uses. That is also the reason why the _get_parameter_table_name method has a different name as the method _get_target_table_name which it calls

**Pull request closed without merging**